appservice/src/siteFiles.ts

```diff
@@ -64,11 +65,31 @@ async function getFsResponse(context: IActionContext, site: ParsedSite, filePath
                 filePath = path.posix.join(linuxHome, filePath);
             }
 
-            const client: ServiceClient = await createGenericClient(context, site.subscription);
-            return await client.sendRequest({
-                method: 'GET',
-                url: `${site.id}/hostruntime/admin/vfs/${filePath}/?api-version=2018-11-01`
-            });
+            /*
+                * Related to issue: https://github.com/microsoft/vscode-azurefunctions/issues/3337
+                Sometimes get a 'BadGateway' or 'ServiceUnavailable' error on initial fetch, but consecutive re-fetching usually fixes the issue.
+                Under these circumstances, we will attempt to do the call 3 times during warmup before throwing the error
+            */
+            let attempts: number = 1;
```

Comment:
Do we not have a common retry class/function? If not we should make one.

Reply:
Oh actually I think I see where we've used the `p-retry` library, should I just re-implement it with that instead?

Example:
![image](https://user-images.githubusercontent.com/40250218/190714281-46c2e2e4-f4df-4828-8ad0-c20a21c1db71.png)

Reply:
Hmm, actually, I don't know if I like using the `p-retry` library's pattern here.  I'm currently retrying only on certain error instances, and then throwing the error else... 
using `p-retry`, I would have to wrap the error as a special `AbortError` to get it to cancel early... not sure if we want to mess with wrapping our errors like that just to get this to work.

I could explore creating our own custom utility version of this if needed, what do you think is the right move here?